pkg/authentication/middleware.go

```diff
@@ -0,0 +1,50 @@
+package authentication
+
+import (
+	"context"
+	"net/http"
+)
+
+type Interceptor[T Ctx] struct {
+	authenticator *Authenticator[T]
+}
+
+func Middleware[T Ctx](authenticator *Authenticator[T]) *Interceptor[T] {
+	return &Interceptor[T]{
+		authenticator: authenticator,
+	}
+}
+
+// RequireAuthentication will check if there is a valid session and provide it in the context.
+// If there is no session, it will automatically start a new authentication (by redirecting the user to the Login UI)
+func (i *Interceptor[T]) RequireAuthentication() func(next http.Handler) http.Handler {
```

Comment:
this is intended, so that if needed we can add a variadic options parameter and with this signature, it's easy to use in middlewares as they typically require `func(next http.Handler) http.Handler`